lib/apiaria.rb

```diff
@@ -28,4 +28,27 @@ def get_mentor_availability(mentor_id)
     JSON.parse(response.body)
   end
   
+  def get_messages(pageNum = nil)
+    unless pageNum
+      response = self.class.get("/message_threads",  headers: { "authorization" => @auth_token })
+      JSON.parse(response.body)
+    else
+      url = "/message_threads?page=" + pageNum.to_s
+      response = self.class.get(url,  headers: { "authorization" => @auth_token })
+      JSON.parse(response.body)
+    end
+  end
+
+  def create_message(sender, recipientId, body, options = {})
```

Comment:
@wonderwoman13 I agree the above approach looks clean but a few things  -

-  _token_ is actually the message thread token per the API doc, not the _@auth_token_
-  How do you account for the optional parameters in the above case? Lets say if a user decides not to pass one or all of the optional parameters(token, subject) when invoking the method?My only reason to use the options hash was to handle the optional parameters with logic in ruby.
-  Lastly,my final create_message API invocation looks exactly like above ,except with the optional parameters included as well.

I will give a try with the above solution,and verify if I can get consistent results.